frontend/src/app/survey/detail/response/response.component.ts

```diff
@@ -1,17 +1,111 @@
-import { Component, OnInit } from '@angular/core';
+import {Component, EventEmitter, Input, OnInit, Output} from '@angular/core';
+import {FormBuilder, FormControl, FormGroup, Validators} from '@angular/forms';
+import {TimeslotCreationDto} from '../../../api/models/timeslot-creation-dto';
+import {ResponseValueDto} from '../../../api/models/response-value-dto';
+import {TimeslotDto} from '../../../api/models/timeslot-dto';
 
 @Component({
-  selector: 'app-response',
-  templateUrl: './response.component.html',
-  styleUrls: ['./response.component.sass']
+    selector: 'app-response',
+    templateUrl: './response.component.html',
+    styleUrls: ['./response.component.sass']
 })
 export class ResponseComponent implements OnInit {
-    isEdit = true;
+    form: FormGroup;
     isNew = false;
+    date: Date = new Date();
+    @Input() timeSlot: TimeslotDto = {id: 0, start: '', end: ''};
+    @Input() timeSlotCreate: TimeslotCreationDto = {start: '', end: ''};
+    @Input() mode: Mode = 'view';
+    @Output() responseEdited = new EventEmitter<TimeslotDto>();
+    @Output() timeslotCreated = new EventEmitter<TimeslotCreationDto>();
+    @Output() timeslotUpdated = new EventEmitter<TimeslotCreationDto>();
+    @Output() deleted = new EventEmitter();
+    @Output() responded = new EventEmitter<ResponseValueDto>();
 
-    constructor() { }
+    constructor(private formBuilder: FormBuilder) {
+        const dateRegex = '^[0-9]{4}\\-((0[1-9]|1[0-2])){1}-(0[1-9]|1[0-9]|2[0-9]|3[0-1])\\' +
```

Comment:
I don't think that a regex is the best way to validate the input. Especially because we should use the build in date tools of JS.